Add base react files

Blu3spirits opened the pull request:
TODO
Currently the root page loads after ~4s when loading, but not
rendering all content. One way to make this more performant is to
adjust the serializers in the backend to have all the needed values
beforehand.

Blu3spirits committed:
Add base react files

- TODO: Currently the root page loads after ~4s when loading, but not
  rendering all content. One way to make this more performant is to
adjust the serializers in the backend to have all needed values
beforehand.

Blu3spirits committed:
Attempt npm linting fix

Blu3spirits committed:
Attempt to fix global vs. relative installs

https://gitter.im/eslint/eslint?at=5d9f6740e8de6f3ca073b3d2

Blu3spirits committed:
Fix lint warning

Blu3spirits committed:
Modify backend to suit front-end needs

Blu3spirits committed:
Change up the loading mechanism of courses

Blu3spirits committed:
Fix a traceback issue where props aren't loaded before users can search

Blu3spirits committed:
Refactor courses serializer and courses

Refactoring in order to better serve and search information

Kiro47 commented on webapp/src/ui/components/coursesList.js:
```diff
@@ -0,0 +1,57 @@
+  { id: 'mtu_equiv.mtu_credits', label: 'Credits'},
+  { id: 'transfer_course_college_code.college_name', label: 'transfer college'},
+  { id: 'transfer_course_state_code.state_name', label: 'transfer state'},
+  { id: 'transfer_course_credit', label: 'transfer course credit'},
```
Why are some of the labels lowercase and others capitalized?

Blu3spirits committed:
Misc improvments

- Drastically improved load time of API call
- Improved search function
- Added pagination to courses table

Kiro47 commented on webapp/src/ui/components/coursesList.js:
```diff
@@ -17,12 +18,45 @@ const columns = [
+          count={total}
+          page={page}
+          onChangePage={this.handleChangePage}
+          onChangeRowsPerPage={this.handleChangeRowsPerPage}
```
All of this but `data` is kind of redundant.

Blu3spirits committed:
Remove redundant variable storage
